simulator/func_sim/operation.h

```diff
@@ -213,16 +210,16 @@ void Datapath<T>::load( const T& value)
     assert( is_load());
     if ( is_unsigned_load())
     {
-        v_dst = value;
+        set_v_dst( value, 0);
     }
     else switch ( get_mem_size())
     {
-        case 0: break; // e.g. fences
-        case 1: v_dst = sign_extension<8>( value); break;
-        case 2: v_dst = sign_extension<16>( value); break;
-        case 4: v_dst = sign_extension<32>( value); break;
-        case 8: v_dst = sign_extension<64>( value); break;
-        case 16: v_dst = sign_extension<128>( value); break;
+        case 0:  break; // e.g. fences
+        case 1:  set_v_dst( sign_extension<8>(value),   0); break;
+        case 2:  set_v_dst( sign_extension<16>(value),  0); break;
+        case 4:  set_v_dst( sign_extension<32>(value),  0); break;
+        case 8:  set_v_dst( sign_extension<64>(value),  0); break;
+        case 16: set_v_dst( sign_extension<128>(value), 0); break;
         default: assert( false);
```

Comment:
Could be separated into two functions.

Reply:
You mean signed/unsigned load?

Reply:
I meant value generation and `set_v_dst`. It should end up is something like:

```
set_v_dst( is_unsigned_load() ? value : sign_extend_for_load( value));
```

Reply:
That's a good opportunity to have a "dynamic" sign extension in `macro.h`:

```c++
static constexpr inline T sign_extension( T value, size_t value)
```